assembl/static2/js/app/components/debate/thread/postActions.jsx

```diff
@@ -97,71 +92,52 @@ class PostActions extends React.Component {
-                  );
-                }
-                if (sentimentCounts[sentiment] > 0 && sentiment === 'disagree') {
-                  return (
-                    <div className="min-sentiment" key={index} style={{ left: `${(count += 1 * 6)}px` }}>
-                      <Disagree size={15} />
-                    </div>
-                  );
-                }
-                if (sentimentCounts[sentiment] > 0 && sentiment === 'dontUnderstand') {
-                  return (
-                    <div className="min-sentiment" key={index} style={{ left: `${(count += 1 * 6)}px` }}>
-                      <DontUnderstand size={15} />
-                    </div>
-                  );
-                }
-                if (sentimentCounts[sentiment] > 0 && sentiment === 'moreInfo') {
-                  return (
-                    <div className="min-sentiment" key={index} style={{ left: `${(count += 1 * 6)}px` }}>
-                      <MoreInfo size={15} />
-                    </div>
-                  );
-                }
-                return null;
-              })}
-            </div>
-            <div className="txt">
-              {this.state.screenWidth >= MEDIUM_SCREEN_WIDTH
-                ? totalSentimentsCount
-                : <Translate value="debate.thread.numberOfReactions" count={totalSentimentsCount} />}
+          <OverlayTrigger overlay={getSentimentStats(totalSentimentsCount, sentimentCounts)} placement="right">
+            <div className="sentiments-count margin-m">
+              <div>
+                {Object.keys(sentimentCounts).map((sentiment, index) => {
+                  if (sentimentCounts[sentiment] > 0 && sentiment === 'like') {
+                    return (
+                      <div className="min-sentiment" key={index} style={{ left: `${(count += 1 * 6)}px` }}>
+                        <Like size={15} />
+                      </div>
+                    );
+                  }
+                  if (sentimentCounts[sentiment] > 0 && sentiment === 'disagree') {
+                    return (
+                      <div className="min-sentiment" key={index} style={{ left: `${(count += 1 * 6)}px` }}>
+                        <Disagree size={15} />
+                      </div>
+                    );
+                  }
+                  if (sentimentCounts[sentiment] > 0 && sentiment === 'dontUnderstand') {
+                    return (
+                      <div className="min-sentiment" key={index} style={{ left: `${(count += 1 * 6)}px` }}>
+                        <DontUnderstand size={15} />
+                      </div>
+                    );
+                  }
+                  if (sentimentCounts[sentiment] > 0 && sentiment === 'moreInfo') {
+                    return (
+                      <div className="min-sentiment" key={index} style={{ left: `${(count += 1 * 6)}px` }}>
+                        <MoreInfo size={15} />
+                      </div>
```

Comment:
refactoring this itches me ^^

Reply:
I agree, just keep the list of sentiments in order and for each render something like : 

```
<div className="min-sentiment" key={index} style={{ left: `${(count += 1 * 6)}px` }}>
   <SvgComponent size={15} />
</div>
```

Reply:
I'm trying to do it but I have a "react array element missing key" error while the divs seems to have keys :/

Reply:
Oh, there seems to be a 'should have a unique "key" prop' error in postActions even if I comment out this part